src/test/integration/integration-test.py

```diff
@@ -21,18 +21,54 @@ def setup(cls, config_path, openapi_path):
     def cleanup(cls):
         cls.session.close()
 
-    # Test case: GET /spendingauthority with valid authority onids
     def test_get_authority_onids(self, endpoint='/spendingauthority'):
         for onid in self.test_cases['valid_authority_onids']:
             params = {'onid': onid}
             response = utils.make_request(self, endpoint, 200, params=params)
-            spending_schema = utils.get_resource_schema(
-                self, 'SpendingAuthorityResource'
-            )
-            utils.check_schema(self, response, spending_schema)
-            response_data = response.json()['data']
-            actual_onid = response_data['id']
-            self.assertEqual(actual_onid.lower(), onid.lower())
+            self.checking_response_schema(response, onid)
+            self.checking_spending_limits(response)
+
+    # Test case: GET /spendingauthority with valid authority onids
+    def checking_response_schema(self, response, onid):
+        spending_schema = utils.get_resource_schema(
+            self, 'SpendingAuthorityResource'
+        )
+        utils.check_schema(self, response, spending_schema)
+        response_data = response.json()['data']
+        actual_onid = response_data['id']
+        logging.debug(f'Request to made to {onid.lower()}')
+        logging.debug(f'Response for {actual_onid.lower()}')
+        self.assertEqual(actual_onid.lower(), onid.lower())
+
+    # Test case: GET /spendingauthority spending limits
+    def checking_spending_limits(self, response):
+        # Checking that the lists of spending limits and indexes are unique
+        # Checking valid onids with spending authority only
+        # To make sure that attributes is not empty
+        response_data = response.json()['data']
```

Comment:
Again, if `response_data` is the only portion of `response` object will be used for these two testing function, it'll make sense if you only fetch it once in `test_get_authority_onids` and pass it into these two functions.